File: test/fixtures/python-project/services/user_service.py
from typing import List, Optional
from models.user import User, AdminUser
from config import DATABASE_URL

class UserService:
    def __init__(self):
        self.db_url = DATABASE_URL
        self._cache = {}
    
    def get_all(self) -> List[User]:
        return []
    
    def get_by_id(self, user_id: int) -> Optional[User]:
        return self._cache.get(user_id)
    
    def create_admin(self, name: str, email: str) -> AdminUser:
        admin = AdminUser(id=0, name=name, email=email, role="admin")
        return admin

File: test/fixtures/javascript-project/services/userService.js
const { User, AdminUser } = require('../models/User');
const config = require('../config');

class UserService {
  constructor(dbUrl) {
    this.dbUrl = dbUrl || config.DATABASE_URL;
    this._cache = new Map();
  }

  async getAll() {
    return Array.from(this._cache.values());
  }

  async getById(id) {
    return this._cache.get(id) || null;
  }

  async create(name, email) {
    const id = this._cache.size + 1;
    const user = new User(id, name, email);
    this._cache.set(id, user);
    return user;
  }

  async createAdmin(name, email, permissions) {
    const id = this._cache.size + 1;
    const admin = new AdminUser(id, name, email, permissions);
    this._cache.set(id, admin);
    return admin;
  }
}

module.exports = { UserService };
